libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2750,8 +2750,16 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			);
 
 		if (!funType->bound())
-			if (auto contractType = dynamic_cast<ContractType const*>(exprType))
-				requiredLookup = contractType->isSuper() ? VirtualLookup::Super : VirtualLookup::Virtual;
+		{
+			if (auto typeType = dynamic_cast<TypeType const*>(exprType))
+			{
+				auto contractType = dynamic_cast<ContractType const*>(typeType->actualType());
+				if (contractType && contractType->isSuper())
+					requiredLookup = VirtualLookup::Super;
+			}
+			else if (dynamic_cast<ContractType const*>(exprType))
+				requiredLookup = VirtualLookup::Virtual;
```

Comment:
Could change this `else if` to `else` and assert the condition.

Reply:
Apparently test fails if you do this.

Reply:
Actually isn't "virtual" wrong here?

Reply:
Why?

Reply:
Because it is a member access and that should always be static. I think in this case, we are doing an external call, where virtual looking doesn't even make sense.

Reply:
I changed it to `Static`. This is normally triggered in expressions such as `this.f`. I'll also change it in develop.